esmvalcore/preprocessor/_regrid.py

```diff
@@ -681,22 +681,36 @@ def regrid(cube, target_grid, scheme, lat_offset=True, lon_offset=True):
     return cube
 
 
-def _rechunk(cube, target_grid):
+def _rechunk(
+    cube: iris.cube.Cube,
+    target_grid: iris.cube.Cube,
+) -> iris.cube.Cube:
     """Re-chunk cube with optimal chunk sizes for target grid."""
     if not cube.has_lazy_data() or cube.ndim < 3:
         # Only rechunk lazy multidimensional data
         return cube
 
-    if 2 * np.prod(cube.shape[-2:]) > np.prod(target_grid.shape):
+    lon_coord = target_grid.coord(axis='X')
+    lat_coord = target_grid.coord(axis='Y')
+    if lon_coord.ndim != 1 or lat_coord.ndim != 1:
+        # This function only supports 1D lat/lon coordinates.
+        return cube
+
+    lon_dim, = target_grid.coord_dims(lon_coord)
+    lat_dim, = target_grid.coord_dims(lat_coord)
+    grid_indices = sorted((lon_dim, lat_dim))
+    target_grid_shape = tuple(target_grid.shape[i] for i in grid_indices)
+
+    if 2 * np.prod(cube.shape[-2:]) > np.prod(target_grid_shape):
```

Comment:
All the new code above seems to be here to allow for the target_grid to have x and y dimensions at arbitrary places. Is my reading correct? In that case, should we afford cube the same possibility, or perhaps it's overkill also for target_grid?

Reply:
that's a good point! alas, this was in the ancient regrid module eons ago, so why not?

Reply:
Currently, the target grid can be an arbitrary NetCDF file, while `cube` is CMORized, so I would expect it to have lat/lon as the last two dimensions.